tox.ini

```diff
@@ -27,6 +27,7 @@ deps =
     pytest-sugar
     isort
     black
+    numpy==1.26.4
```

Comment:
Should we pin this in `pyproject.toml` instead of here?

Reply:
its in there as well!

Reply:
In there it's not hard pinned. Just a lower bound. I wonder if it's worth hard pinning it.

Reply:
We have avoided doing this in the past (I think mainly on my request?) but I wonder if it's worth starting to do (in our installation docs we do recommend using a venv etc...)...

We shouldn't delay merging this though, @miller-ian's contribution is a super helpful bug fix that lets other PRs get through :)